nvme-wrap.c

```diff
@@ -20,8 +20,8 @@
  */
 #define do_admin_op(op, d, ...) ({					\
 	int __rc;							\
-	if (d->type == NVME_DEV_DIRECT)					\
-		__rc = nvme_ ## op(d->direct.fd, __VA_ARGS__);		\
+	if (d->type == NVME_DEV_DIRECT || d->type == NVME_DEV_XNVME)	\
```

Comment:
The complete `nvme-wrap.c` file shouldn't be needed after introducing the transport abstraction.